src/twisted/trial/test/test_tests.py

```diff
@@ -174,8 +174,8 @@ def test_results(self):
         self.assertTrue(self.reporter.wasSuccessful())
         self.assertEqual(self.reporter.errors, [])
         self.assertEqual(self.reporter.failures, [])
-        self.assertEqual(len(self.reporter.skips), 3)
-        self.assertEqual(self.reporter.successes, 0)
+        self.assertEqual(len(self.reporter.skips), 2)
+        self.assertEqual(self.reporter.successes, 3)
```

Comment:
This was really confusing until I looked at what the implementation actually does and what it does is *not* skip a test that has a skip attribute set to `""`.  Isn't this the opposite of the goal of the PR?

Reply:
Thanks for the comment. The title of the PR was wrong.
The idea is to handle the empty string as don't skip...
so that you can have

```
if SOMEMTHING:
  skip = 'Shere reason'
else:
  skip = ""
```

Instead of the following code that will trigger a mypy warning.

```
if SOMEMTHING:
  skip = 'Shere reason'
else:
  skip = None
```